### PR TITLE
langchain[patch]: Allow for passing bytestore into multi vector retriever

### PR DESCRIPTION
**Pull request by bracesproul**

### CODE REVIEW

File: examples/src/retrievers/multi_vector_hypothetical.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { InMemoryStore } from "langchain/storage/in_memory";
 import { TextLoader } from "langchain/document_loaders/fs/text";
 import { Document } from "langchain/document";
 import { JsonKeyOutputFunctionsParser } from "langchain/output_parsers";
+import { createDocumentStoreFromByteStore } from "langchain/storage/encoder_backed";
```
Comment (jacoblee93) [inline examples/src/retrievers/multi_vector_hypothetical.ts:15]: The whole idea here was to never see this function again?

File: examples/src/retrievers/multi_vector_hypothetical.ts [hunk 2]
```diff
@@ -89,9 +90,11 @@ const keyValuePairs: [string, Document][] = docs.map((originalDoc, i) => [
...
 ]);
 
-// The docstore to use to store the original chunks
-const docstore = new InMemoryStore();
-await docstore.mset(keyValuePairs);
+// The byteStore to use to store the original chunks
+const byteStore = new InMemoryStore<Uint8Array>();
+// Convert the byteStore to a docStore to use for retrieval
+const docStore = createDocumentStoreFromByteStore(byteStore);
+await docStore.mset(keyValuePairs);
```
Comment (jacoblee93) [inline examples/src/retrievers/multi_vector_hypothetical.ts:97]: Examples should use `retriever.docstore.mset` or we could maybe add something like `retriever.addDocuments`

File: examples/src/retrievers/multi_vector_hypothetical.ts (hunk 1 above)
Comment (bracesproul) [reply examples/src/retrievers/multi_vector_hypothetical.ts:15]: Whoops, was unaware. Will fix.

File: examples/src/retrievers/multi_vector_hypothetical.ts (hunk 2 above)
Comment (bracesproul) [reply examples/src/retrievers/multi_vector_hypothetical.ts:97]: I like the `addDocuments` method, will implement

File: langchain/src/retrievers/multi_vector.ts [hunk 3]
```diff
@@ -3,15 +3,20 @@ import {
...
 import { Document } from "../document.js";
+import { createDocumentStoreFromByteStore } from "../storage/encoder_backed.js";
 
 /**
  * Arguments for the MultiVectorRetriever class.
  */
 export interface MultiVectorRetrieverInput extends BaseRetrieverInput {
   vectorstore: VectorStoreInterface;
-  docstore: BaseStoreInterface<string, Document>;
+  /** @deprecated Prefer `baseStore`. */
```
Comment (jacoblee93) [inline langchain/src/retrievers/multi_vector.ts:17]: `byteStore`

File: langchain/src/retrievers/multi_vector.ts [hunk 4]
```diff
@@ -55,12 +60,44 @@ export class MultiVectorRetriever extends BaseRetriever {
...
+        "byteStore and docstore are undefined. Please provide at least one."
+      );
+    }
     this.idKey = args.idKey ?? "doc_id";
     this.childK = args.childK;
     this.parentK = args.parentK;
   }
 
+  /**
+   * Add documents to the byte store.
```
Comment (jacoblee93) [inline langchain/src/retrievers/multi_vector.ts:78]: This should also add index them in the vectorstore too

If it doesn't make sense to do so, we can leave it as `retriever.docstore.add()` above.

**Pull request merged**